packages/rpc-core/src/types/rpc-methods/common.d.ts

```diff
@@ -0,0 +1,6 @@
+// TODO: Eventually move this into whatever package implements transactions
+declare type Finality = 'confirmed' | 'finalized' | 'processed';
+
+declare type Slot =
+    // TODO(solana-labs/solana/issues/30341) Represent as bigint
+    number;
```

Comment:
So, fast forward to when everything is `bigint` (or rewind, and have represented all `u64s` as JavaScript strings in the first place) and all that the transport has to do is to serialize `bigints` as `string` over the wire.

We can't do this today, because the Solana RPC doesn't accept them.

```shell
# params is normally [5, 10] but here I'm representing them as ["5", "10"]
curl https://api.mainnet-beta.solana.com -X POST -H "Content-Type: application/json" -d '
  {
    "jsonrpc": "2.0", "id": 1,
    "method": "getBlocks",
    "params": ["5", "10"]
  }
'
{"jsonrpc":"2.0","error":{"code":-32602,"message":"Invalid params: invalid type: string \"5\", expected u64."},"id":1}
```

We need to change the server to parse strings as `u64` before we can do that on the client.

Reply:
But I see what you're saying. You're saying: make it a `bigint` now at the outer part of the interface, and have the transport (temporarily) downcast that to a JavaScript `number`, maybe optionally warning when it overflows. Is that right?

Reply:
I love this!

Reply:
I was wondering it if it would be possible at the transport level to use `bigint` everywhere, but I forgot that the problem is at the RPC level, and not web3.js. But yeah, since all of these interfaces are new, it might be worth doing everything right and use `bigint`s everywhere.